### DISCUSSION TITLE
At what point are the plugins, like pyenv, called from the plugins list in the .zshrc file?

### DISCUSSION THREAD

**Discussion by Disco-Chef** [ohmyzsh/ohmyzsh#9930] [Help] [answered]
I would have thought that the file is read line by line and executes them as such (i.e. it reaches that `plugins = (...)` line and calls/loads up each of the plugins), but I am unsure.
Perhaps the whole file is read, the variables are created and only after that the plugins are actually called/loaded?

Practically speaking, I am just asking so as to know whether I should set PATH variables for pyenv before the `plugins = ( ... pyenv ....)` line, or whether i can do it after that line (if the plugins are loaded after the whole `.zshrc` is read)

**Comment by mcornella** [comment, marked as answer]
They aren't loaded until Oh My Zsh is sourced, in the `source $ZSH/oh-my-zsh.sh` line. We recommend setting `PATH` before everything else in your zshrc file though, just for a cleaner zshrc file.

**Reply by Disco-Chef** [reply]
Awesome! Thank you!
